pkg/pod/pod_test.go

```diff
@@ -76,3 +79,237 @@ func TestWithCustomResourcesRequests(t *testing.T) {
+		testBuilder, err := buildPodTestBuilderWithFakeObjects(runtimeObjects, testCase.pod.Name, testCase.pod.Namespace)
+		assert.Nil(t, err)
+
+		err = testBuilder.WaitUntilHealthy(2*time.Second, testCase.includeSucceeded, testCase.checkReadiness,
+			testCase.ignoreFailedPods)
+
+		assert.Equal(t, testCase.expectedErrMsg, getErrorString(err))
+	}
+}
+
+func TestWaitForAllPodsInNamespacesHealthy(t *testing.T) {
+	testCases := []struct {
+		namespaces       []string
+		includeSucceeded bool
+		checkReadiness   bool
+		ignoreFailedPods bool
+		ignoreNamespaces []string
+		expectedErrMsg   string
+	}{
+		{
+			namespaces:       []string{"ns1"},
+			includeSucceeded: true,
+			checkReadiness:   false,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "",
+		},
+		{
+			namespaces:       []string{"ns1"},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "",
+		},
+		{
+			namespaces:       []string{"ns2"},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "context deadline exceeded",
+		},
+		{
+			namespaces:       []string{""},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "context deadline exceeded",
+		},
+		{
+			namespaces:       []string{""},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{"ns2"},
+			expectedErrMsg:   "context deadline exceeded",
+		},
+	}
+
+	var runtimeObjects []runtime.Object
+	runtimeObjects = append(runtimeObjects, generateTestPod("test1", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test2", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test3", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test4", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test5", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test1", "ns2", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test2", "ns2", corev1.PodRunning, corev1.PodInitialized,
+		false))
+	fakeClient := k8sfake.NewSimpleClientset(runtimeObjects...)
+
+	for _, testCase := range testCases {
+		err := WaitForAllPodsInNamespacesHealthy(&clients.Settings{
+			K8sClient:       fakeClient,
+			CoreV1Interface: fakeClient.CoreV1()}, testCase.namespaces, 2*time.Second, testCase.includeSucceeded,
+			testCase.checkReadiness,
+			testCase.ignoreFailedPods, testCase.ignoreNamespaces)
+
+		assert.Equal(t, testCase.expectedErrMsg, getErrorString(err))
+	}
+}
```

Comment:
It seems like this file already doesn't, but the helper functions should be removed to all be below the tests

Reply:
fixed